src/plugins/autocompletion_providers/Git.ts

```diff
@@ -84,181 +141,310 @@ const notStagedFiles = unique(async(context: AutocompletionContext): Promise<Sug
     }
 });
 
-const subCommandProviders: Dictionary<AutocompletionProvider> = {
-    add: combine([notStagedFiles, addOptions]),
-    checkout: combine([branchesExceptCurrent, branchAlias, notStagedFiles]),
-    commit: commitOptions,
-    status: statusOptions,
-    merge: combine([branchesExceptCurrent, branchAlias]),
-    push: pushOptions,
-    config: combine([configVariables, configOptions]),
-};
+interface GitCommandData {
+    name: string;
+    description: string;
+    provider?: AutocompletionProvider;
+    aliases: string[];
+}
 
-const subCommands = [
+const subCommandsData: GitCommandData[] = [
     {
         name: "add",
         description: "Add file contents to the index.",
+        provider: combine([notStagedFiles, addOptions]),
+        aliases: [],
```

Comment:
Why do you need to associate aliases with particular subcommands? It seems like it doesn't matter much.